### PR TITLE
Issue196

### PR DESCRIPTION
**Pull request by mayabrandi**
### This PR adds a script to get values from csv and put the values on artifact udf in the step.

**How to prepare for test**:
- [x] Install on climical-lims-stage:
```
conda activate cg_lims
pip install -U "git+https://github.com/Clinical-Genomics/cg_lims@issue196"
```

- [x] configure the commands in epps:

```
lims -c <config> epps -l <log> -p <process id>files csv-well-to-udf -u 'Size (bp)' -vf 'Average Size [bp]' -wf 'WellId' -f 'Tapestation CSV'
lims -c <config> epps -l <log> -p <process id>files csv-well-to-udf -u 'DIN Value' -vf 'DIN' -wf 'Well' -f 'Tapestation DIN CSV'
```

- [x] activate the epps in the tapestation step and make sure the file placeholders: `Tapestation DIN CSV` and `Tapestation CSV` exist

### How to test.
- [x] place some samples in the step and uoppload the two files, run the scripts and check that they work.

### Before deply.
- [x] make sure other steps that are using the same EPP are having the correct file placeholder configured.

### Review:
- [x] Code approved by @Mropat 
- [x] Tests executed by @mayabrandi  @cesve 
- [x] "Merge and deploy" approved by @mayabrandi 

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by cesve** [comment]
The step has been tested and worked well. 
Screenshot 2021-01-22 at 09 26 30

**Comment by cesve** [comment]
Files used in the test:

[rc200925tga-gDNA_sampleTable DIN-values.xlsx](https://github.com/Clinical-Genomics/cg_lims/files/5854474/rc200925tga-gDNA_sampleTable.DIN-values.xlsx)
[RC201016tga-gDNA_compactRegionTable Size-values.xlsx](https://github.com/Clinical-Genomics/cg_lims/files/5854483/RC201016tga-gDNA_compactRegionTable.Size-values.xlsx)

File: cg_lims/get/files.py [hunk 1]
```diff
@@ -27,3 +27,19 @@ def get_log_content(lims: Lims, file_id: str) -> str:
         log_file_content = ''
 
     return log_file_content
+
+
+def get_result_file(process:Process, file_path: Optional[str]=None, file_name: Optional[str]=None):
+    """Reads file from args if present. Otherwise searches for file in outarts with
```
Comment (Mropat) [inline cg_lims/get/files.py:33]: outarts = output artifacts?

File: cg_lims/get/files.py [hunk 2]
```diff
@@ -27,3 +27,19 @@ def get_log_content(lims: Lims, file_id: str) -> str:
         log_file_content = ''
 
     return log_file_content
+
+
+def get_result_file(process:Process, file_path: Optional[str]=None, file_name: Optional[str]=None):
```
Comment (Mropat) [inline cg_lims/get/files.py:32]:
```suggestion
def get_result_file(process:Process, file_path: Optional[str]=None, file_name: Optional[str]=None) -> str:
```

File: cg_lims/get/files.py [hunk 3]
```diff
@@ -27,3 +27,19 @@ def get_log_content(lims: Lims, file_id: str) -> str:
...
+    """Reads file from args if present. Otherwise searches for file in outarts with
+     name <file_name>"""
+
+    if file_path and Path(file_path).is_file():
+         file_path = file_path
```
Comment (Mropat) [inline cg_lims/get/files.py:37]:
```suggestion
         return file_path
```

File: cg_lims/get/files.py [hunk 4]
```diff
@@ -27,3 +27,19 @@ def get_log_content(lims: Lims, file_id: str) -> str:
...
+    else:
+        files = list(filter(lambda a: a.name in [file_name], process.all_outputs()))
```
Comment (Mropat) [inline cg_lims/get/files.py:39]: This variable name is files, but what it is, is an output. Maybe re-name it to make less confusing to understand what the list contains? Or maybe immediately get files[0] from each output in the list? That way row 43  would bemore understandable.

File: cg_lims/get/files.py [hunk 5]
```diff
@@ -27,3 +27,19 @@ def get_log_content(lims: Lims, file_id: str) -> str:
...
+        if len(files)>1:
+            raise FileError(f'more than one file named {file_name}')
+        else:
+            file_path = files[0].files[0].content_location.split('scilifelab.se')[1]
```
Comment (Mropat) [inline cg_lims/get/files.py:43]:
```suggestion
           return files[0].files[0].content_location.split('scilifelab.se')[1]
```

File: cg_lims/EPPs/files/file_to_udf.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+@click.command()
+@options.file_placeholder(help="File placeholder name.")
+@options.local_file()
+@options.udf()
+@options.well_field()
+@options.value_field()
+@options.input()
+@click.pass_context
+def csv_well_to_udf(ctx, file, well_field, value_field, udf, input, local_file):
```
Comment (Mropat) [inline cg_lims/EPPs/files/file_to_udf.py:69]: type hints? :yum:

File: cg_lims/EPPs/files/file_to_udf.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def set_udfs(well_field: str, value_field: str, udf: str, well_dict: dict, result_file: Path):
+    """Reads the csv and sets the value for each sample"""
+
+    failed_arts = False
+    passed_arts = 0
+    csv_reader = pd.read_csv(result_file, encoding='latin1')
+    data = csv_reader.transpose().to_dict()
+    for i, sample in data.items():
```
Comment (Mropat) [inline cg_lims/EPPs/files/file_to_udf.py:44]:
```suggestion
    for _, sample in data.items():
```

this way you can send the unused variable into the void

File: cg_lims/get/files.py (hunk 2 above)
Comment (mayabrandi) [reply cg_lims/get/files.py:32]: 👍

File: cg_lims/get/files.py (hunk 1 above)
Comment (mayabrandi) [reply cg_lims/get/files.py:33]: 👍

File: cg_lims/get/files.py (hunk 4 above)
Comment (mayabrandi) [reply cg_lims/get/files.py:39]: Im rewriting to make it cleaner. Thanks for noticing

**Pull request merged**